src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/DWMColorization.cs

```diff
@@ -0,0 +1,280 @@
+
+        if(err != 0) 
+        {
+            primaryAccent = secondaryAccent = tertiaryAccent = systemAccent;
+        }
+        else
+        {
+            uint Accent1, Accent2, Accent3;
+
+            if (isDarkTheme)
+            {
+                Accent1 = GetColorFromPreference(ref colorPreference, IMMERSIVE_COLOR_TYPE.IMCLR_SystemAccentDark1, false, IMMERSIVE_HC_CACHE_MODE.IHCM_REFRESH);
+                Accent2 = GetColorFromPreference(ref colorPreference, IMMERSIVE_COLOR_TYPE.IMCLR_SystemAccentDark2, false, IMMERSIVE_HC_CACHE_MODE.IHCM_REFRESH);
+                Accent3 = GetColorFromPreference(ref colorPreference, IMMERSIVE_COLOR_TYPE.IMCLR_SystemAccentDark3, false, IMMERSIVE_HC_CACHE_MODE.IHCM_REFRESH);
+            }
+            else
+            {
+                Accent1 = GetColorFromPreference(ref colorPreference, IMMERSIVE_COLOR_TYPE.IMCLR_SystemAccentLight1, false, IMMERSIVE_HC_CACHE_MODE.IHCM_REFRESH);
+                Accent2 = GetColorFromPreference(ref colorPreference, IMMERSIVE_COLOR_TYPE.IMCLR_SystemAccentLight2, false, IMMERSIVE_HC_CACHE_MODE.IHCM_REFRESH);
+                Accent3 = GetColorFromPreference(ref colorPreference, IMMERSIVE_COLOR_TYPE.IMCLR_SystemAccentLight3, false, IMMERSIVE_HC_CACHE_MODE.IHCM_REFRESH);
+            }
+
+            primaryAccent = Color.FromArgb(0xff, (byte)Accent1, (byte)(Accent1 >> 8), (byte)(Accent1 >> 16));
+            secondaryAccent = Color.FromArgb(0xff, (byte)Accent2, (byte)(Accent2 >> 8), (byte)(Accent2 >> 16));
+            tertiaryAccent = Color.FromArgb(0xff, (byte)Accent3, (byte)(Accent3 >> 8), (byte)(Accent3 >> 16));
+        }
+
+        UpdateColorResources(systemAccent, primaryAccent, secondaryAccent, tertiaryAccent);
+
+        _currentApplicationAccentColor = systemAccent;
+    }
+
+    /// <summary>
+    /// Updates application resources.
+    /// </summary>        
+    private static void UpdateColorResources(
+        Color systemAccent,
+        Color primaryAccent,
+        Color secondaryAccent,
+        Color tertiaryAccent)
+    {
+#if DEBUG
+        System.Diagnostics.Debug.WriteLine("INFO | SystemAccentColor: " + systemAccent, "System.Windows.Accent");
+        System
+            .Diagnostics
+            .Debug
+            .WriteLine("INFO | SystemAccentColorPrimary: " + primaryAccent, "System.Windows.Accent");
+        System
+            .Diagnostics
+            .Debug
+            .WriteLine("INFO | SystemAccentColorSecondary: " + secondaryAccent, "System.Windows.Accent");
+        System
+            .Diagnostics
+            .Debug
+            .WriteLine("INFO | SystemAccentColorTertiary: " + tertiaryAccent, "System.Windows.Accent");
+#endif
+
+        if (ThemeColorization.IsThemeDark())
+        {
+#if DEBUG
+            System.Diagnostics.Debug.WriteLine("INFO | Text on accent is DARK", "System.Windows.Accent");
+#endif
+            Application.Current.Resources["TextOnAccentFillColorPrimary"] = Color.FromArgb(
```

Comment:
Won't work in case of window level theming.

Reply:
Currently working on that, but I was of the opinion that we fix that along with the implementation of API proposal for enabling/disabling theme via property setting at application and window level.

Reply:
It will crash if there is no Application.Current and it's the completely wrong approach to do dynamic resource management for themes.
So using this as the starting point for the other changes doesn't feel right.
It requires a lot more work till this PR is ready to be merged.
As i said in a previous comment i have concerns regarding the quality of the changes in this PR.